packages/familie-backend/src/auth/bruker.ts

```diff
@@ -12,7 +16,63 @@ export const hentBrukerprofil = () => {
             email: req.session.upn,
             groups: req.session.groups,
             identifier: req.session.upn,
+            enhet: req.session.enhet,
         };
         res.status(200).send(user);
     };
 };
+
+// Hent brukerenhet
+export const hentBrukerenhet = (saksbehandlerTokenConfig: ITokenRequest) => {
+    
+    const msGraphOBOTokenConfig: ITokenRequest = {
+        clientId: saksbehandlerTokenConfig.clientId,
+        clientSecret: saksbehandlerTokenConfig.clientSecret,
+        redirectUrl: saksbehandlerTokenConfig.redirectUrl,
+        scope: `https://graph.microsoft.com/.default`,
+        tokenUri: saksbehandlerTokenConfig.tokenUri,
+    };
+    
+    return async (req:Request, _: Response, next: NextFunction) => {
+
+        const msGraphMeUrl= `https://graph.microsoft.com/v1.0/me`
+        if(!req.session){
+            throw new Error('Mangler sesjon på kall');
+        }
+
+        if(req.session.enhet){
+            logInfo(req, "Gyldig enhet i session");
+            return next();
+        }
+
+        logInfo(req, "Enhet ikke i session, henter fra "+ msGraphMeUrl);
+        
+        const obotoken= await hentOnBehalfOfToken(req, saksbehandlerTokenConfig, msGraphOBOTokenConfig);
+
+        request
+        .get(
+            { url: msGraphMeUrl, headers:{
+                Authorization: 'Bearer '+ obotoken,
+            } },
+            (_err, _httpResponse, body) => {
+                return body;
+            },
+        )
+        .then(result => {
+            const enhet= JSON.parse(result)?.officeLocation?.slice(0, 4);
+            if(!enhet){
+                throw new Error('officeLocation ikke funnet fra Microsoft Graph, eller formatet er uventet');
+            }
+            
+            if(!req.session){
```

Comment:
I guess we dont need this since we have already checked it at line 43?

Reply:
theoretically we need the check because the session can be lost between line 43 and the line (asynchronous call of request.get). If we skip the check, a compile error will be thrown due to the req.session can be null.